src/codecs/encoding/transformer.rs

```diff
@@ -128,11 +128,23 @@ impl Transformer {
 
     fn apply_only_fields(&self, log: &mut LogEvent) {
         if let Some(only_fields) = self.only_fields.as_ref() {
-            let old_value = std::mem::replace(log.value_mut(), Value::Object(BTreeMap::new()));
+            let mut old_value = std::mem::replace(log.value_mut(), Value::Object(BTreeMap::new()));
 
             for field in only_fields {
-                if let Some(value) = old_value.get(field) {
-                    log.insert((PathPrefix::Event, field), value.clone());
+                if let Some(value) = old_value.remove(field, true) {
+                    log.insert((PathPrefix::Event, field), value);
+                }
```

Comment:
Is this get -> remove change just an optimization, it doesn't seem related to the functional change?

Reply:
1. It removes the need to `clone` the value when inserting into the new `log`.
2. By removing it from `old_value` it indicates that the field has been added to the log, so below when we are getting the value meaning `service` it won't exist in `old_value` and thus we don't need to add it to `dropped_fields`.

It's a beautiful part of Rust, because there is no `clone` this field can only exist at most once - either in the log message or in the `dropped_fields`.